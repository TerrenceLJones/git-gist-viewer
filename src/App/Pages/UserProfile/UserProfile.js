import React from 'react';

const UserProfile = ({ user, onClickBackToResults }) => {
    if (!user) return '';
    const imgStyles = {
        borderRadius: '5px',
        height: '100px',
        width: '100px'
    };

    return (
        <div style={{margin: 'auto', width: '500px'}}>
            <button onClick={onClickBackToResults}>&lt;</button>
            <h1>{user.name}</h1>
            <img src={`${user.avatar_url}`} style={imgStyles} alt="" />
            <div>{user.userName}</div>
            <div>{user.company}</div>
            <div>{user.email}</div>
            <div>{user.bio}</div>
        </div>
    );
};

export default UserProfile;
