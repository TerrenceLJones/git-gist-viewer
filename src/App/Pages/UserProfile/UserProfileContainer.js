import React from 'react';
import { compose, lifecycle, withHandlers } from 'recompose';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import _ from 'lodash';

import { nonOptimalStates } from 'redux/utils/nonOptimalStates';
import { usersActions, usersTypes } from 'redux/modules/users';
import { apiSelectors } from 'redux/modules/api'
import UserProfile from './UserProfile';

const userLoadingSelector = apiSelectors.createLoadingSelector([usersTypes.API_HANDLER_LOAD_USER]);
const userErrorSelector = apiSelectors.createErrorMessageSelector([usersTypes.API_HANDLER_LOAD_USER]);


const LoadingDisplay = () => "Loading...";
const ErrorDisplay = ({ error }) => error;
const UserProfileContainer = ({ onClickBackToResults, user }) => {
    return <UserProfile user={user} onClickBackToResults={onClickBackToResults} />;
}

const mapStateToProps = (state, ownProps) => {
    const { match: { params: { userName } } } = ownProps;
    const user = state.users.selectedUser;

    return {
        isLoading: user ? userLoadingSelector(state) : true,
        loadError: userErrorSelector(state),
        user: user,
        userName
    }
};

const mapDispatchToProps = dispatch => ({
    loadUser: userName => dispatch(usersActions.loadUserRequest(userName))
});

const enhance = compose(
    connect(mapStateToProps, mapDispatchToProps),
    withRouter,
    lifecycle({
        componentDidMount() {
            const { userName } = this.props
            this.props.loadUser(userName);
        },
        componentDidUpdate(prevProps) {
            const prevUserName = _.get(prevProps, 'match.params.userName');
            const currUserName = _.get(this.props, 'match.params.userName');

            if (currUserName !== prevUserName) {
                this.props.loadUser(currUserName);
            }
        }
    }),
    withHandlers({
        onClickBackToResults: ({ history }) => () => {
            history.goBack();
        }
    }),
    nonOptimalStates([
        { when: ({ loadError }) => !!loadError, render: ErrorDisplay },
        { when: ({ isLoading }) => isLoading, render: LoadingDisplay }
    ])
);

export default enhance(UserProfileContainer);
