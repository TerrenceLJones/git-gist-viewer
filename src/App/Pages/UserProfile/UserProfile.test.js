import React from 'react';
import { shallow } from 'enzyme';
import UserProfile from './UserProfile';

it('renders without crashing', () => {
    expect(shallow(<UserProfile user={{}} />)).toMatchSnapshot();
});
