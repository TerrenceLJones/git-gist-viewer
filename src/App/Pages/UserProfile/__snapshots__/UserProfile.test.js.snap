// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserProfile
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          onClick={undefined}
>
          &lt;
</button>,
        <h1 />,
        <img
          alt=""
          src="undefined"
          style={
                    Object {
                              "borderRadius": "5px",
                              "height": "100px",
                              "width": "100px",
                            }
          }
/>,
        <div />,
        <div />,
        <div />,
        <div />,
      ],
      "style": Object {
        "margin": "auto",
        "width": "500px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "<",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "<",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "src": "undefined",
          "style": Object {
            "borderRadius": "5px",
            "height": "100px",
            "width": "100px",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            onClick={undefined}
>
            &lt;
</button>,
          <h1 />,
          <img
            alt=""
            src="undefined"
            style={
                        Object {
                                    "borderRadius": "5px",
                                    "height": "100px",
                                    "width": "100px",
                                  }
            }
/>,
          <div />,
          <div />,
          <div />,
          <div />,
        ],
        "style": Object {
          "margin": "auto",
          "width": "500px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "<",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "<",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "undefined",
            "style": Object {
              "borderRadius": "5px",
              "height": "100px",
              "width": "100px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
